File: 11-container-with-most-water/11-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int max = 0;
        int start = 0;
        int end = height.length - 1;
        while (start < end) {
            max = Math.max(Math.min(height[start], height[end]) * (end - start), max);
            if (height[start] < height[end]) {
                start++;
            } else {
                end--;
            }
        }

        return max;
    }
}

File: 11-container-with-most-water/11-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        result = 0
        start = 0
        end = len(height) - 1
        
        while start < end:
            result = max(min(height[start], height[end]) * (end - start), result)
            
            if height[start] < height[end]:
                start += 1
            else:
                end -= 1

        return result